src/Dhall.hs

```diff
@@ -282,12 +298,21 @@ inputWithSettings
     -- ^ The decoded value in Haskell
 inputWithSettings settings (Type {..}) txt = do
     expr  <- throws (Dhall.Parser.exprFromText (view sourceName settings) txt)
-    expr' <- Dhall.Import.loadDirWith
-               (view rootDirectory settings)
-               Dhall.Import.exprFromImport
-               (view startingContext settings)
-               (Dhall.Core.getReifiedNormalizer (view normalizer settings))
-               expr
+
+    let InputSettings {..} = settings
+
+    let EvaluateSettings {..} = _evaluateSettings
+
+    let transform =
+               set Dhall.Import.protocolVersion _protocolVersion
+            .  set Dhall.Import.normalizer      _normalizer
+            .  set Dhall.Import.startingContext _startingContext
+            .  set Dhall.Import.rootDirectory   _rootDirectory
+
+    let status = transform (Dhall.Import.emptyStatus ".")
```

Comment:
This doesn't look right. I am not sure why `Status` has grown the `_rootDirectory` field since it's not actually being used (?), but, when that gets set, it doesn't do the right thing since the root import is always going to be relative to `.` since that's what's passed to `emptyStatus`.

Note that the test suite isn't exercising this code, and that the test suite is invoking `emptyStatus` with the root directory rather than `.`.

Reply:
Whoops, good catch.  I'll fix this